cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -244,13 +256,14 @@ func (s *server) refreshExistingUTXOs() error {
 		return err
 	}
 
-	s.utxos = make(map[externalapi.DomainOutpoint]*walletUTXO, len(getUTXOsByAddressesResponse.Entries))
+	s.utxosSortedByAmount = make([]*walletUTXO, 0, len(getUTXOsByAddressesResponse.Entries))
```

Comment:
Binary search above is especially important here since it will reduce the overall insertion cost from n^2 to nlogn